### PR TITLE
feat: add PR label logic

### PR DESCRIPTION
**Pull request by rahalan**
Checks new PRs and adds labels, depending if module owner can approve, or core team is needed

### CODE REVIEW

File: .github/workflows/avm.platform.set-avm-github-pr-labels.yml
Comment (eriqua) [inline .github/workflows/avm.platform.set-avm-github-pr-labels.yml:5]: Does this work for PRs open from a fork?

File: .github/workflows/avm.platform.set-avm-github-pr-labels.yml
Comment (eriqua) [inline .github/workflows/avm.platform.set-avm-github-pr-labels.yml:34]: I guess this should only run in upstream.

File: .github/workflows/avm.platform.set-avm-github-pr-labels.yml
Comment (rahalan) [reply .github/workflows/avm.platform.set-avm-github-pr-labels.yml]: I guess so

File: .github/workflows/avm.platform.set-avm-github-pr-labels.yml
Comment (rahalan) [reply .github/workflows/avm.platform.set-avm-github-pr-labels.yml]: @eriqua I added a condition now, that it can only run in Azure/bicep-registry-modules repo & I renamed the file (removed the avm. part)

File: .github/workflows/avm.platform.set-avm-github-pr-labels.yml
Comment (eriqua) [reply .github/workflows/avm.platform.set-avm-github-pr-labels.yml]: I think this trigger only works for PRs open from a branch, but I may be wrong

Comment (rahalan) [reply .github/workflows/avm.platform.set-avm-github-pr-labels.yml]: @eriqua that is a repository setting, see https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#enabling-workflows-for-forks-of-private-repositories

Comment (eriqua) [reply .github/workflows/avm.platform.set-avm-github-pr-labels.yml]: Not sure I follow, the provided link is mentioning private repositories

File: avm/utilities/pipelines/platform/Set-AvmGitHubPrLabels.ps1
Comment (eriqua) [inline avm/utilities/pipelines/platform/Set-AvmGitHubPrLabels.ps1:40]: Due to latest changes to codeowners file
```suggestion
    $teamNames = [array](Get-GithubPrRequestedReviewerTeamNames -PrUrl $PrUrl.Replace('api.', '').Replace('repos/', '') | Where-Object { $_ -ne 'bicep-admins' -and $_ -ne 'avm-core-team-technical-bicep' -and $_ -ne 'avm-module-reviewers-bicep' })
```

File: avm/utilities/pipelines/platform/Set-AvmGitHubPrLabels.ps1
Comment (eriqua) [inline avm/utilities/pipelines/platform/Set-AvmGitHubPrLabels.ps1:52]: This is a first step and already helping a lot. We miss 2 cases, your call if including them here or take them in a subsequent PR:
- orphaned modules: check if the module is orphaned and apply both the needs core team and the module orphaned label to the PR
- new modules: they fall into this latest condition, but they cannot be reviewed by module owners, since related PRs also modify the issue template and the codeowners file. Also, in general, the core team is needed if anything but the module folder is modified. E.g. is someone where to update psrule suppressions, that would require the core team. Luckily, the new codeowners structure will help us. The core team label has to be applied whenever the bicep team or the core team is added as reviewer. In fact, these are added only if ci related files are updated.

File: .github/workflows/platform.set-avm-github-pr-labels.yml
Comment (eriqua) [inline .github/workflows/platform.set-avm-github-pr-labels.yml:20]: @rahalan since I'm not confident enough on how this external action treats secrets, and given this pipeline is expected to run from forks, so anyone can trigger it, I'd appreciate a second opinion on the workflow itself

File: .github/workflows/avm.platform.set-avm-github-pr-labels.yml
Comment (rahalan) [reply .github/workflows/avm.platform.set-avm-github-pr-labels.yml]: @eriqua you are right, my bad. Overall a great catch from your side. I changed the trigger to pull_request_target, see https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target

File: avm/utilities/pipelines/platform/Set-AvmGitHubPrLabels.ps1
Comment (rahalan) [reply avm/utilities/pipelines/platform/Set-AvmGitHubPrLabels.ps1]: done

File: .github/workflows/platform.set-avm-github-pr-labels.yml
Comment (rahalan) [reply .github/workflows/platform.set-avm-github-pr-labels.yml:20]: @ChrisSidebotham can you have a look? Thanks.

File: avm/utilities/pipelines/platform/Set-AvmGitHubPrLabels.ps1
Comment (eriqua) [inline avm/utilities/pipelines/platform/Set-AvmGitHubPrLabels.ps1:45]: Unless I'm missing something I don't think teamMembers is initialized at this point of the script

File: avm/utilities/pipelines/platform/Set-AvmGitHubPrLabels.ps1
Comment (eriqua) [reply avm/utilities/pipelines/platform/Set-AvmGitHubPrLabels.ps1:72]: Looks good for new modules. For the orphaned one I think something is off with the order or variable names

File: avm/utilities/pipelines/platform/Set-AvmGitHubPrLabels.ps1
Comment (eriqua) [inline avm/utilities/pipelines/platform/Set-AvmGitHubPrLabels.ps1:47]: We are not binary anymore :) we will also have utl modules

File: avm/utilities/pipelines/platform/Set-AvmGitHubPrLabels.ps1
Comment (eriqua) [inline avm/utilities/pipelines/platform/Set-AvmGitHubPrLabels.ps1:49]: If I get it correctly we are retrieving the whole list of modules from the csv and doing a replace and comparison operation for each. I think the cost of this should be optimized. I'd suggest to instead retrieve the module name from the team name and just do one operation in checking if that team status is orphaned

File: avm/utilities/pipelines/platform/Set-AvmGitHubPrLabels.ps1
Comment (rahalan) [reply avm/utilities/pipelines/platform/Set-AvmGitHubPrLabels.ps1:47]: That means then also rework on the other logic. Thanks for sharing

File: avm/utilities/pipelines/platform/Set-AvmGitHubPrLabels.ps1
Comment (rahalan) [reply avm/utilities/pipelines/platform/Set-AvmGitHubPrLabels.ps1:49]: As we do not use Voyager 2 and have extreme limitations, I think we should not care. We are talking about operations that take milliseconds. Joke aside, what do you have in mind?

File: avm/utilities/pipelines/platform/Set-AvmGitHubPrLabels.ps1
Comment (rahalan) [reply avm/utilities/pipelines/platform/Set-AvmGitHubPrLabels.ps1:45]: Good catch. Stupid copy paste. Let me refine it

Comment (rahalan) [reply avm/utilities/pipelines/platform/Set-AvmGitHubPrLabels.ps1]: Done

File: avm/utilities/pipelines/platform/Set-AvmGitHubPrLabels.ps1
Comment (eriqua) [reply avm/utilities/pipelines/platform/Set-AvmGitHubPrLabels.ps1:61]: Fair, in case we can improve later

File: avm/utilities/pipelines/platform/Set-AvmGitHubPrLabels.ps1
Comment (eriqua) [reply avm/utilities/pipelines/platform/Set-AvmGitHubPrLabels.ps1:59]: Let's limit the addition to this PR only. We'll update the other logic later to facilitate this review

File: avm/utilities/pipelines/platform/Set-AvmGitHubPrLabels.ps1
Comment (eriqua) [inline avm/utilities/pipelines/platform/Set-AvmGitHubPrLabels.ps1:58]: Is this assignment correct? We are assigning the first element of team **members** to the module name. Did you mean teamNames[0]?

File: .github/workflows/platform.set-avm-github-pr-labels.yml
Comment (ChrisSidebotham) [reply .github/workflows/platform.set-avm-github-pr-labels.yml:20]: @eriqua - Not sure what the concern is here, you would need access to the secrets to leverage the job & token. This is a standard external task which is used by CAE in other repo's too @jtracey93 can confirm

File: avm/utilities/pipelines/platform/Set-AvmGitHubPrLabels.ps1
Comment (rahalan) [reply avm/utilities/pipelines/platform/Set-AvmGitHubPrLabels.ps1]: @eriqua good catch. I fixed it now.

File: avm/utilities/pipelines/platform/Set-AvmGitHubPrLabels.ps1
Comment (rahalan) [reply avm/utilities/pipelines/platform/Set-AvmGitHubPrLabels.ps1:59]: @eriqua agree. Can you please share the third option, so I can check for that one? Thanks.

Comment (rahalan) [reply avm/utilities/pipelines/platform/Set-AvmGitHubPrLabels.ps1]: @eriqua done. Also updated the function to retrieve CSV files (now supports utility modules)

File: avm/utilities/pipelines/platform/Set-AvmGitHubPrLabels.ps1
Comment (eriqua) [inline avm/utilities/pipelines/platform/Set-AvmGitHubPrLabels.ps1:21]: Suggest to stay generic to avoid getting stale (it's already outdated name)
```suggestion
Will be triggered by the corresponding platform workflow
```

File: avm/utilities/pipelines/platform/Set-AvmGitHubPrLabels.ps1
Comment (eriqua) [inline avm/utilities/pipelines/platform/Set-AvmGitHubPrLabels.ps1:69]: This seems to be out of the foreach loop of row 63 so `$module.` should not be accessible

Comment (rahalan) [reply avm/utilities/pipelines/platform/Set-AvmGitHubPrLabels.ps1]: @eriqua fixed. Although PS somehow still preserved the variable, because it worked.

Comment (eriqua) [reply avm/utilities/pipelines/platform/Set-AvmGitHubPrLabels.ps1]: Looks good. I see you removed the break all together. To optimize I'd suggest to keep it but move it after the assignment. Once that is done no need to complete the iteration

File: .github/workflows/platform.set-avm-github-pr-labels.yml
Comment (jtracey93) [reply .github/workflows/platform.set-avm-github-pr-labels.yml:20]: yup you need access to the private key which is in a secret in this repo.

I thought this repo was only meant to run on this repo so if so add the if condition to ensure that

File: avm/utilities/pipelines/platform/Set-AvmGitHubPrLabels.ps1
Comment (rahalan) [reply avm/utilities/pipelines/platform/Set-AvmGitHubPrLabels.ps1]: done

File: .github/workflows/platform.set-avm-github-pr-labels.yml
Comment (rahalan) [reply .github/workflows/platform.set-avm-github-pr-labels.yml:20]: @jtracey93 that check is already in the PR, see 
image

Comment (eriqua) [reply .github/workflows/platform.set-avm-github-pr-labels.yml:20]: Thanks @ChrisSidebotham and @jtracey93 for confirming and @rahalan for your patience! Approving

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**
